build-advisories/src/Roave/SecurityAdvisories/VersionConstraint.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Roave\SecurityAdvisories;
+
+/**
+ * A simple version constraint - naively assumes that it is only about ranges like ">=1.2.3,<4.5.6"
+ */
+final class VersionConstraint
+{
+    const CLOSED_RANGE_MATCHER    = '^>(=?)\s*((\d+.)*\d+)\s*,\s*<(=?)\s*((\d+.)*\d+)$';
+    const LEFT_OPEN_RANGE_MATCHER = '^<(=?)\s*((\d+.)*\d+)$';
+    const RIGHT_OPEN_RANGE_MATCHER = '^>(=?)\s*((\d+.)*\d+)$';
```

Comment:
ranges without upper bounds are forbidden in the security advisory database (the upper bound is the version in which you fixed the security issue): https://github.com/FriendsOfPHP/security-advisories/blob/513cffa670f376be46353888ce6afaf67200fbcd/validator.php#L115